keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1361 @@
+}
+
+// DeviceCapacity describes a quantity associated with a device.
+type DeviceCapacity struct {
+   // Value defines how much of a certain device capacity is available.
+   //
+   // +required
+   Value resource.Quantity `json:"value" protobuf:"bytes,1,rep,name=value"`
+
+
+   // ClaimPolicy specifies that this device capacity must be consumed
+   // by each resource claim according to the defined consumption policy.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   ClaimPolicy *CapacityClaimPolicy `json:"claimPolicy,omitempty" protobuf:"bytes,2,opt,name=claimPolicy"`
+}
+
+// CapacityClaimPolicy defines consumption policy for consumable capacity.
+// Either one of the consumption policies must be defined.
+type CapacityClaimPolicy struct {
+   // Set defines a set of acceptable quantities of consuming requests.
+   //
+   // +optional
+   // +oneOf=CapacityClaimPolicy
+   Set *CapacityClaimPolicySet `json:"set,omitempty" protobuf:"bytes,1,opt,name=set"`
+
+
+   // Range defines an acceptable quantity range of consuming requests.
+   // +optional
+   // +oneOf=CapacityClaimPolicy
+   Range *CapacityClaimPolicyRange `json:"range,omitempty" protobuf:"bytes,2,opt,name=range"`
+}
+
+
+// CapacityClaimPolicySet defines a discrete set of allowable capacity values for consumption.
+type CapacityClaimPolicySet struct {
+   // Default specifies the default capacity to be used for a consumption request
+   // if no value is explicitly provided.
+   //
+   // +required
+   Default resource.Quantity `json:"default,omitempty" protobuf:"bytes,1,name=default"`
+
+
+   // Options defines a discrete set of additional valid capacity values that can be requested.
+   // +optional
+   // +listType=atomic
+   Options []resource.Quantity `json:"options,omitempty" protobuf:"bytes,2,rep,name=options"`
+}
+
+
+// CapacityClaimPolicyRange defines a valid range of consuming capacity.
+type CapacityClaimPolicyRange struct {
+   // Minimum specifies the minimum capacity required for a consumption request.
+   // This field is required and serves as the default capacity to be consumed.
+   //
+   // +required
+   Minimum resource.Quantity `json:"minimum,omitempty" protobuf:"bytes,1,name=minimum"`
+
+
+   // Maximum specifies the maximum capacity that can be requested in a consumption request.
+   //
+   // +optional
+   Maximum *resource.Quantity `json:"maximum,omitempty" protobuf:"bytes,2,opt,name=maximum"`
+
+
+   // Step defines the incremental block size by which capacity can increase from the minimum.
+   //
+   // +optional
+   Step *resource.Quantity `json:"step,omitempty" protobuf:"bytes,3,opt,name=step"`
+}
+
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+
+type DeviceRequest struct {
+...
+   // Capacities define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   Capacities *CapacityRequirements `json:"capacities,omitempty" protobuf:"bytes,9,opt,name=capacities"`
```

Comment:
Perhaps call this "CapacityRequests" in analogy to https://github.com/kubernetes/kubernetes/blob/680ea07dbb2c6050d13b93660fa4d27d2d28d6eb/staging/src/k8s.io/api/core/v1/types.go#L2696?

Just "Requests" would be misleading because we already have such a field.

Or "Required"?